fluent-react/src/provider.js

```diff
@@ -21,57 +22,40 @@ import createParseMarkup from "./markup";
  * `ReactLocalization` to format translations.  If a translation is missing in
  * one instance, `ReactLocalization` will fall back to the next one.
  */
-export default class LocalizationProvider extends Component {
-  constructor(props) {
-    super(props);
-    const {bundles, parseMarkup} = props;
-
-    if (bundles === undefined) {
-      throw new Error("LocalizationProvider must receive the bundles prop.");
-    }
-
-    if (!bundles[Symbol.iterator]) {
-      throw new Error("The bundles prop must be an iterable.");
-    }
-
-    this.l10n = new ReactLocalization(bundles);
-    this.parseMarkup = parseMarkup || createParseMarkup();
+function LocalizationProvider(props) {
+  if (props.bundles === undefined) {
+    throw new Error("LocalizationProvider must receive the bundles prop.");
   }
 
-  getChildContext() {
-    return {
-      l10n: this.l10n,
-      parseMarkup: this.parseMarkup,
-    };
+  if (!props.bundles[Symbol.iterator]) {
+    throw new Error("The bundles prop must be an iterable.");
   }
 
-  componentWillReceiveProps(next) {
-    const { bundles } = next;
-
-    if (bundles !== this.props.bundles) {
-      this.l10n.setBundles(bundles);
-    }
-  }
-
-  render() {
-    return Children.only(this.props.children);
-  }
+  return createElement(
+    FluentContext.Provider,
+    { value: {
+      l10n: new ReactLocalization(props.bundles, props.parseMarkup),
+      parseMarkup: props.parseMarkup || createParseMarkup()
```

Comment:
I'm not sure about this one yet, as I don't see a semantic reason for them to live together. Are there other ergonomic/performance considerations that might make us want to do that?

Reply:
I'm mostly basing this on how this works in `@fluent/dom`. There we have the [`DOMLocalization`](https://github.com/projectfluent/fluent.js/blob/master/fluent-dom/src/dom_localization.js) class which has all the logic for localizing DOM elements. I'd lean towards considering `parseMarkup` part of `ReactLocalization`, too.

In longer term, I expect `parseMakrup` to be removed as a configurable property, which would leave `l10n` as the only property of the context value. I don't see many benefits to keeping this extra level of nesting, but perhaps it's because I generally tend to prefer flatter structures :)